src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -213,10 +218,8 @@ public void start(final Map<String, String> props) {
               BsonDocument keyDocument =
                   new BsonDocument(ID_FIELD, changeStreamDocument.get(ID_FIELD));
 
-              SchemaAndValue keySchemaAndValue =
-                  getSchemaAndValue(sourceConfig.getKeyOutputFormat(), keyDocument);
-              SchemaAndValue valueSchemaAndValue =
-                  getSchemaAndValue(sourceConfig.getValueOutputFormat(), valueDoc);
+              SchemaAndValue keySchemaAndValue = keySchemaAndValueProducer.get(keyDocument);
```

Comment:
Shouldn't here be used `valueDoc` or `changeStreamDocument` instead of `keyDocument `? Because now using any scheme except `DEFAULT_AVRO_KEY_SCHEMA` doesn't make much sense.

Reply:
:+1: Thanks ortex - thats a good catch.  I'll also add a test to ensure alternative key format support.

Reply:
Can you explain this change a little more?  Why did the default key schema work with just the id field, but no other schema would?

Reply:
Now with schema you can decide what the record key field should contain, so it needs the full change stream document to be able to do that.  This actually fixes: KAFKA-123 as now users can use schemas to define the record key